data/summary/types/test/summary.go

```diff
@@ -184,49 +231,78 @@ func RandomBGMSummary(userId string) *types.Summary[types.BGMStats, *types.BGMSt
 		}
 	}
 
+	//for i := 0; i < len(datum.Stats.DailyBuckets); i++ {
+	//	datum.Stats.DailyBuckets[i] = types.Bucket[types.BGMBucketData, *types.BGMBucketData]{
+	//		Date:           test.RandomTime(),
+	//		LastRecordTime: test.RandomTime(),
+	//		Data: &types.BGMBucketData{
+	//			TargetRecords:   test.RandomIntFromRange(0, 288),
+	//			LowRecords:      test.RandomIntFromRange(0, 288),
+	//			VeryLowRecords:  test.RandomIntFromRange(0, 288),
+	//			HighRecords:     test.RandomIntFromRange(0, 288),
+	//			VeryHighRecords: test.RandomIntFromRange(0, 288),
+	//			TotalGlucose:    test.RandomFloat64FromRange(0, 10000),
+	//			TotalRecords:    test.RandomIntFromRange(0, 288),
+	//		},
+	//	}
+	//}
+
 	for _, period := range []string{"1d", "7d", "14d", "30d"} {
 		datum.Stats.Periods[period] = &types.BGMPeriod{
+			HasAverageGlucose: test.RandomBool(),
 			AverageGlucose: &types.Glucose{
 				Value: test.RandomFloat64FromRange(1, 30),
 				Units: glucose.MmolL,
 			},
-			HasAverageGlucose: test.RandomBool(),
+			AverageGlucoseDelta: pointer.FromAny(test.RandomFloat64FromRange(1, 30)),
```

Comment:
Ditto here

Reply:
It was moved to the top of this area as it did not match the ordering of the following variables.